tdp/core/collection.py

```diff
@@ -37,128 +37,74 @@ class Collection:
     """An enriched version of an Ansible collection.
 
     A TDP Collection is a directory containing playbooks, DAGs, default variables and schema variables.
-
-    Attributes:
-        path: Path to the collection.
-        name: Name of the collection.
-        dag_directory: Path to the DAG directory.
-        default_vars_directory: Path to the default variables directory.
-        operations_directory: Path to the operations directory.
-        schema_directory: Path to the schema variables directory.
-        dag_yamls: List of DAG files in the YAML format.
-        operations: Dictionary of operations with the operation name as key and the operation path as value.
     """
 
     def __init__(self, path: Union[str, os.PathLike]):
-        """Initialize a collection from a path.
+        """Initialize a collection.
 
         Args:
             path: The path to the collection.
         """
-        self._path = Path(path)
+        self._path = Path(path).expanduser().resolve()
+        self._validate_path()
         self._dag_yamls: Optional[List[Path]] = None
         self._operations: Optional[Dict[str, Path]] = None
-
-    @staticmethod
-    def from_path(path: Union[str, os.PathLike]):
-        """Factory method to build a collection from a path.
-
-        Args:
-            path: The path to the collection.
-
-        Returns:
-            A collection.
-
-        Raises:
-            PathDoesNotExistsError: If the path does not exists.
-            PathIsNotADirectoryError: If the path is not a directory.
-            MissingMandatoryDirectoryError: If the path does not contain the mandatory directories.
-        """
-        path = Path(path).expanduser().resolve()
-        if not path.exists():
-            raise PathDoesNotExistsError(f"{path} does not exists")
-        if not path.is_dir():
-            raise PathIsNotADirectoryError(f"{path} is not a directory")
-        for mandatory_directory in MANDATORY_DIRECTORIES:
-            mandatory_path = path / mandatory_directory
-            if not mandatory_path.exists() or not mandatory_path.is_dir():
-                raise MissingMandatoryDirectoryError(
-                    f"{path} does not contain the mandatory directory {mandatory_directory}",
-                )
-        return Collection(path)
-
-    @property
-    def path(self) -> Path:
-        """Path to the collection."""
-        return self._path
+        self._services: Optional[List[Path]] = None
 
     @property
     def name(self) -> str:
         """Name of the collection."""
         return self._path.name
 
-    @property
-    def dag_directory(self) -> Path:
```

Comment:
Why removing theses methods ? Just because it's not used outside the class doesn't mean it should be deleted.